Makefile

```diff
@@ -102,7 +102,7 @@ prepare-e2e: kuttl set-test-image-vars set-image-controller container
 	$(KUSTOMIZE) build config/crd -o tests/_build/crds/
 
 set-test-image-vars:
-	IMG=local/opentelemetry-operator:e2e
+	$(eval IMG=local/opentelemetry-operator:e2e)
```

Comment:
Eval is typically not a good solution within Makefiles, but given that this is only used for setting up tests, it should be an okay solution for the time being.

Reply:
Could you try running the release workflow locally without and with this change? I don't really understand this fix (vs. the current behavior), but there are a few things about Makefiles that I don't understand anyway :-)

Reply:
@jpkrohling - Confirmed. Below are the two cases to be aware of:

1. Without any changes to the make file, the release process uses the "local/opentelemetry-operator:e2e" image.
![Screen Shot 2021-01-23 at 1 33 33 PM](https://user-images.githubusercontent.com/1246157/105613614-d70bbd80-5d80-11eb-94e7-5252e67e4414.png)

2. With `IMG=local/opentelemetry-operator:e2e` indented under `set-test-image-vars`, the release artifacts correctly use the quay.io image; however, now the e2e workflow is broken https://github.com/open-telemetry/opentelemetry-operator/blob/e23224ee92ac86d9b31d70dc580a25c1883d2d9d/.github/workflows/e2e.yaml#L21 as `IMG` is still set to the global quay.io image, since the assignment of `IMG` within `set-test-image-vars` does not change the value of `IMG` for the reset of the target prerequisites (set-image-controller container).
![Screen Shot 2021-01-23 at 1 34 05 PM](https://user-images.githubusercontent.com/1246157/105613621-e25ee900-5d80-11eb-9828-6a7cad6038b9.png)
![Screen Shot 2021-01-23 at 1 35 57 PM](https://user-images.githubusercontent.com/1246157/105613629-ef7bd800-5d80-11eb-9e78-77c4b6e239b8.png)


With this PR, both of the issues identified above are resolved. However, we may want to revisit portions of this Makefile in the future.

Reply:
I'll merge this and do a dry-run for 0.18.1.